docs/public/specifications/tests/DNSSEC-TP/dnssec03.md

```diff
@@ -1,73 +1,317 @@
-## DNSSEC03: Check for too many NSEC3 iterations
+## DNSSEC03: Verify NSEC3 parameters
+
 
 ### Test case identifier
-**DNSSEC03** Check for too many NSEC3 iterations
+**DNSSEC03**
+
+
+## Table of contents
+
+* [Objective](#objective)
+* [Scope](#scope)
+* [Inputs](#inputs)
+* [Summary](#summary)
+* [Test procedure]
+* [Outcome(s)](#outcomes)
+* [Special procedural requirements](#special-procedural-requirements)
+* [Intercase dependencies](#intercase-dependencies)
+* [Terminology](#terminology)
+
+
+## Objective
+
+The NSEC3 record type and its parameters are defined in [RFC 5155]. The
+recommended values of the parameters have been updated by [RFC 9276].
+
+For NSEC3 there are four fields that determine how the NSEC3 record are created
+and interpreted ([RFC 5155][RFC 5155#section-3], section3):
+
+* Hash algorithm
+* Flags
+* Iterations
+* Salt
+
+**Hash algorithm:** The only legal value of the hash algorithm is value 1
+(SHA-1). See ([RFC 5155][RFC 5155#section-11], section 11 and
+[IANA NSEC3 Parameters registry]).
+
+**Flags:** The only defined flags in the flag field is bit 7, "opt-out". It may
```

Comment:
Depending on whom you ask, the least significant bit of a byte can be numbered 7 or 0. In RFC 5155, bit 7 is the least significant bit. It might be good to remind the reader of the convention that RFC 5155 chose.

Reply:
Added clarification. -- That is the way they always count in the RFC, isn't it?

Reply:
You’re probably right. I never stopped to think about what the usual bit numbering convention is in network protocol specifications, actually. But it never hurts to remind the reader about those conventions.

At my old job, I spent a lot of time perusing old ITU-T recommendations about ISDN such as [Q.931](https://www.itu.int/rec/T-REC-Q.931-199805-I/en). In there, the convention is different: bits are numbered from 1 to 8, where bit 1 is the least-significant bit, and on ISDN networks, bits are transmitted least-significant bit first.

In bitwise arithmetic, it makes sense to number the bits from 0 to 7 where bit 0 is the least-significant bit: that way, bit N is worth 2^N if set. I helped @tgreenx a little bit with the implementation and we both fell into the trap of differing conventions.